src/backend/executor/nodeHashjoin.c

```diff
@@ -2157,3 +2174,225 @@ ExecHashJoinInitializeWorker(HashJoinState *state,
 		ExecSetExecProcNode(&state->js.ps, ExecParallelHashJoin);
 	}
 }
+
+/*
+ * Find "inner var = outer var" in hj->hashclauses and create runtime filter
+ * for it.
+ */
+void
+CreateRuntimeFilter(HashJoinState* hjstate)
+{
+	AttrNumber lattno, rattno;
+	Expr		*expr;
+	JoinType	jointype;
+	HashJoin	*hj;
+	HashState	*hstate;
+	PlanState	*target;
+	AttrFilter	*attr_filter;
+	ListCell	*lc;
+
+	/*
+	 * Only applicatable for inner, right and semi join,
+	 */
```

Comment:
Could you give a little more explain about why these join types are supported and others are not?

Reply:
Add more message to explain why just inner, right and semi join are allowed with runtime filter.
fix it in https://github.com/apache/cloudberry/commit/98dac6dfc7d5e44e111aa16bdf5948d07ee2eb00